src/Components/MainPageComponent/ArticleComments.js

```diff
@@ -1,55 +1,101 @@
 import React from "react";
+import { useState } from "react";
 import PropTypes from "prop-types";
 import { connect } from "react-redux";
 import dateFormat from "dateformat";
+import { onPostCommentsClicked } from "../../ReduxStore/FeedDetails/feedActions";
+
 
 const InternalArticleComments = props => {
+   const [myComment, setMyComment] = useState("");
+  
   return (
-    <div className="row">
-      <div className="col-xs-12 col-md-8 offset-md-2">
-        {props.currentComments.comments &&
-          props.currentComments.comments.map((comment, index) => {              
-            return (
-              <div key={index}>
-                <div className="card">
-                  <div className="card-block">
-                    <p className="card-text">{comment.body}</p>
-                  </div>
-                  <div className="card-footer">
-                    <a href="#top" className="comment-author">
-                      <img
-                        src={comment.author.image}
-                        className="comment-author-img"
-                        alt="au"
-                      />
-                    </a> &nbsp;
-                    <a href="#top" className="comment-author">
-                      {comment.author.username}
-                    </a>
-                    <span className="date-posted">
-                      {dateFormat(
-                        comment.updatedAt,
-                        "ddd mmm dd yyyy"
-                      )}
-                    </span>
+    <div className='commonts-container'>
+
+      {/* ---------------- Your Comments ----------------  */}
+      <div className='row'>
+        <div className='col-xs-12 col-md-8 offset-md-2'>
+          <form className='card comment-form'>
+            <div className='card-block'>
+              <textarea
+                onChange={event => setMyComment(event.target.value)}
+                className='form-control'
+                placeholder='Write a comment...'
+                rows='3'></textarea>
+            </div>
+            <div className='card-footer'>
+              <img
+                src='http://i.imgur.com/Qr71crq.jpg'
+                className='comment-author-img'
+                alt='au'
+              />
+              <button 
```

Comment:
The button without type will by default be the submit button of a form. That's why when you click the button, the post comment API call is always canceled and your comment will never be posted, because by submitting the form it will triggere the page refresh, which will cancel all the on-going requests.